eng/Versions.props

```diff
@@ -22,20 +22,27 @@
     <LastReleasedStableAssemblyVersion>$(AssemblyVersion)</LastReleasedStableAssemblyVersion>
     <UsingToolMicrosoftNetCompilers>true</UsingToolMicrosoftNetCompilers>
   </PropertyGroup>
+  <!--
+    For source generator support we need to target multiple versions of Rolsyn in order to be able to run on older versions of Roslyn.
+    We pin these versions as we need to match them exactly for any scenarios that run Roslyn on .NET Framework, like Visual Studio.
+  -->
   <PropertyGroup>
-    <!-- For source generator support we need to target multiple versions of Rolsyn in order to be able to run on older versions of Roslyn -->
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion_3_11>3.11.0</MicrosoftCodeAnalysisCSharpWorkspacesVersion_3_11>
-    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.0.1</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
-    <MicrosoftCodeAnalysisVersion>4.0.1</MicrosoftCodeAnalysisVersion>
+    <MicrosoftCodeAnalysisVersion_3_11>3.11.0</MicrosoftCodeAnalysisVersion_3_11>
+    <MicrosoftCodeAnalysisVersion_4_0>4.0.1</MicrosoftCodeAnalysisVersion_4_0>
   </PropertyGroup>
   <PropertyGroup>
     <!-- Code analysis dependencies -->
-    <MicrosoftCodeAnalysisAnalyzersVersion>3.3.2</MicrosoftCodeAnalysisAnalyzersVersion>
-    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.0.1</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
-    <MicrosoftCodeAnalysisCSharpVersion>4.0.1</MicrosoftCodeAnalysisCSharpVersion>
+    <MicrosoftCodeAnalysisAnalyzersVersion>3.3.3</MicrosoftCodeAnalysisAnalyzersVersion>
+    <MicrosoftCodeAnalysisCSharpCodeStyleVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCSharpCodeStyleVersion>
+    <MicrosoftCodeAnalysisCSharpWorkspacesVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCSharpWorkspacesVersion>
+    <MicrosoftCodeAnalysisCSharpVersion>4.2.0-2.22128.1</MicrosoftCodeAnalysisCSharpVersion>
```

Comment:
@eerhardt does this need to be a change that makes our OOB source generators cross-compile?  What's going to happen when someone now installs the JSON or logging source generator package on a 6.0 GA SDK/Visual Studio 2022?

Reply:
I specifically updated our in-box analyzers that we have already shipped to be pinned to the RTM versions of the Roslyn API surface they built against when they shipped. The JSON and Logging generators are pinned to 3.11 and 4.0. They don't build against these new versions.

Reply:
Can we guarantee that all `net7.0` consumers will have this version of roslyn?  I know we have some surprising combinations that are sometimes supported due to the roslyn shipping in VS at a different cadence.

Reply:
By the time this commit ships, we will have at least this version of Roslyn in the product. By the time .NET 7 ships, all places where .NET 7 is supported will also have a version of Roslyn that is at least this version (likely the release version of 4.2.0)